python/api/job/serializers.py

```diff
@@ -23,44 +21,61 @@
 import cm.status_api
 import cm.config as config
 from cm.errors import AdcmEx
-from cm.models import JobLog, HostProvider, Host, Cluster, ClusterObject, ServiceComponent
+from cm.models import JobLog, Host, ClusterObject, ServiceComponent, get_object_cluster
 from api.api_views import hlink
 
 
-def get_job_objects(obj):
-    resp = []
-    selector = obj.selector
-    for obj_type in selector:
-        try:
-            if obj_type == 'cluster':
-                cluster = Cluster.objects.get(id=selector[obj_type])
-                name = cluster.name
-            elif obj_type == 'service':
-                service = ClusterObject.objects.get(id=selector[obj_type])
-                name = service.prototype.display_name
-            elif obj_type == 'component':
-                comp = ServiceComponent.objects.get(id=selector[obj_type])
-                name = comp.prototype.display_name
-            elif obj_type == 'provider':
-                provider = HostProvider.objects.get(id=selector[obj_type])
-                name = provider.name
-            elif obj_type == 'host':
-                host = Host.objects.get(id=selector[obj_type])
-                name = host.fqdn
-            else:
-                name = ''
-        except ObjectDoesNotExist:
-            name = 'does not exist'
-        resp.append(
-            {
-                'type': obj_type,
-                'id': selector[obj_type],
-                'name': name,
-            }
-        )
+def get_object_name(obj):
+    if hasattr(obj, 'name'):
+        return obj.name
+    elif hasattr(obj, 'fqdn'):
+        return obj.fqdn
+    else:
+        return obj.prototype.display_name
+
+
+def get_job_objects(task):
+    if not task.task_object:
+        return None
+
+    def cook_obj(obj_type, obj_id, name):
+        return {'type': obj_type, 'id': obj_id, 'name': name}
+
+    obj = task.task_object
+    obj_type = obj.prototype.type
+    resp = [cook_obj(obj_type, obj.id, get_object_name(obj))]
+    if obj_type == 'service':
+        resp.append(cook_obj('cluster', obj.cluster.id, get_object_name(obj.cluster)))
+    elif obj_type == 'component':
+        resp.append(cook_obj('cluster', obj.cluster.id, get_object_name(obj.cluster)))
+        resp.append(cook_obj('service', obj.service.id, get_object_name(obj.service)))
+    elif obj_type == 'host':
+        resp.append(cook_obj('provider', obj.provider.id, get_object_name(obj.provider)))
     return resp
 
 
+def get_task_selector(obj, action):
```

Comment:
Since you removed the function `get_selector()`, you need to correct the new function `get_task_selector()` according to the latest changes. See: https://github.com/arenadata/adcm/pull/886

Reply:
but I already have cluster in get_task_selector()

component = ServiceComponent.obj.get(cluster=cluster, prototype=action.prototype)
service = ClusterObject.obj.get(cluster=cluster, prototype=action.prototype)

Or am I missing somethig else?

Reply:
See comment below